src/crons/update-farm-rewards/update-farm-rewards-for-week.ts

```diff
@@ -1,74 +1,81 @@
 import { DB_DECIMALS } from "../../constants";
-import { getTableConfig } from "drizzle-orm/pg-core";
+
 import { db } from "../../db/db";
-import { getTableColumns } from "drizzle-orm";
-import {
-  FarmDatabaseType,
-  farmRewards,
-  FarmRewardsDatabaseType,
-  farms,
-} from "../../db/schema";
+import { eq } from "drizzle-orm";
+import { farmRewards, FarmRewardsInsertType, farms } from "../../db/schema";
 import { getScrapedFarmsAndRewards } from "./get-scraped-farms-and-rewards-for-week";
-import { sql } from "drizzle-orm";
+
 export async function updateFarmRewardsForWeek({
   weekNumber,
 }: {
   weekNumber: number;
 }) {
-  const farmColumns = getTableColumns(farms);
-  const farmTableConfig = getTableConfig(farms);
   const farmsWithRewards = await getScrapedFarmsAndRewards({ weekNumber });
 
-  const drizzleFarmType: FarmDatabaseType[] = farmsWithRewards.map((farm) => {
-    return {
-      id: farm.hexPubKey,
-      totalGlowRewards:
-        BigInt(Math.floor(farm.rewards.glow)) * BigInt(10 ** DB_DECIMALS),
-      totalUSDGRewards:
-        BigInt(Math.floor(farm.rewards.usdg)) * BigInt(10 ** DB_DECIMALS),
-      // shortId: Number(farm.shortId), // fix the cron by creating devices with shortId and asigning them to farms, get the device id by calling rust api
-      auditCompleteDate: farm.auditCompleteDate
-        ? new Date(Number(farm.auditCompleteDate))
-        : new Date(),
-      createdAt: new Date(),
-      gcaId: farm.installerWallet,
-      userId: farm.payoutWallet,
-    };
+  const allFarmsIdsWithDevices = await db.query.farms.findMany({
+    columns: {
+      id: true,
+      totalGlowRewards: true,
+      totalUSDGRewards: true,
+      oldShortIds: true,
+    },
+    with: {
+      devices: {
+        columns: {
+          shortId: true,
+          publicKey: true,
+        },
+      },
+    },
   });
 
-  const farmWithRewards: FarmRewardsDatabaseType[] = farmsWithRewards.map(
-    (farm) => {
-      return {
-        hexlifiedFarmPubKey: farm.hexPubKey,
-        weekNumber,
-        usdgRewards:
-          BigInt(Math.floor(farm.rewards.usdg)) * BigInt(DB_DECIMALS),
-        glowRewards:
-          BigInt(Math.floor(farm.rewards.glow)) * BigInt(DB_DECIMALS),
-      };
+  const farmWithRewards: (FarmRewardsInsertType & {
+    previousUsdgRewards: bigint;
+    previousGlowRewards: bigint;
+  })[] = farmsWithRewards.reduce((carry: any[], farm) => {
+    const farmMatch = allFarmsIdsWithDevices.find(
+      (dbFarm) =>
+        dbFarm.devices.find((device) => device.shortId === farm.shortId) ||
```

Comment:
We should always filter by hexPubkey, never by short Id. Although unlikely, we can have short id collisions between GCAs.